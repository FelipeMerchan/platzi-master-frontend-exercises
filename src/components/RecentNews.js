import React from 'react'

import './styles/RecentNews.css'

function RecentNews(props) {
  const handleClick = (event) => {
    event.preventDefault()
    props.onClick(event)
  }

  return (
    <section className="RecentNews">
      <h2 className="RecentNews___title">Noticias Recientes</h2>
      <div>
        <h5 className="RecentNews__date">Publicada: 09.09.2012</h5>
        <p className="RecentNews__text">
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
          ullamcorper.
        </p>
      </div>
      <div>
        <h5 className="RecentNews__date">Publicada: 09.09.2012</h5>
        <p className="RecentNews__text">
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
          ullamcorper.
        </p>
      </div>
      <div className="mt-2">
        <a
          className="button"
          role="button"
          href="/"
          onClick={handleClick}
          target="_blank"
          data-button="footer"
        >
          MÃ¡s noticias
        </a>
      </div>
    </section>
  )
}

export default RecentNews