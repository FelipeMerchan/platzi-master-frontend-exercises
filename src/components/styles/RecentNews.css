.RecentNews {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 30%;
  margin-right: 5%;
  font-family: var(--font-3);
}

.RecentNews___title {
  margin-bottom: var(--space);
  padding-bottom: var(--space);
  border-bottom: 1px solid #dfdfdf;
  color: #48504d;
}


.RecentNews__text {
  margin-bottom: var(--space-2);
  line-height: 1.4;
  color: #a6a7a2;
}