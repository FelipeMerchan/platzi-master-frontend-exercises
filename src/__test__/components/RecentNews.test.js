import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import RecentNews from '../../components/RecentNews'

describe('Probar el componente <RecentNews />', () => {
  const recentNews = shallow(<RecentNews />)
  test('Render del componente RecentNews', () => {
    expect(recentNews.length).toEqual(1)
  })
  test('Render del tÃ­tulo', () => {
    expect(recentNews.find('.RecentNews___title').text()).toEqual('Noticias Recientes')
  })
})

describe('RecentNews Snapshot', () => {
  test('Comprobar la UI del componente RecentNews', () => {
    const recentNews = create(<RecentNews />)
    expect(recentNews.toJSON()).toMatchSnapshot()
  })
})