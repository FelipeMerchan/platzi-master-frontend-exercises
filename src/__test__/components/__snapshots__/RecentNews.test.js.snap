// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentNews Snapshot Comprobar la UI del componente RecentNews 1`] = `
<section
  className="RecentNews"
>
  <h2
    className="RecentNews___title"
  >
    Noticias Recientes
  </h2>
  <div>
    <h5
      className="RecentNews__date"
    >
      Publicada: 09.09.2012
    </h5>
    <p
      className="RecentNews__text"
    >
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper.
    </p>
  </div>
  <div>
    <h5
      className="RecentNews__date"
    >
      Publicada: 09.09.2012
    </h5>
    <p
      className="RecentNews__text"
    >
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper.
    </p>
  </div>
  <div
    className="mt-2"
  >
    <a
      className="button"
      data-button="footer"
      href="/"
      onClick={[Function]}
      role="button"
      target="_blank"
    >
      MÃ¡s noticias
    </a>
  </div>
</section>
`;
